rasa/core/channels/mattermost.py

```diff
@@ -132,12 +132,12 @@ def from_credentials(cls, credentials: Optional[Dict[Text, Any]]) -> InputChanne
 
         # pytype: disable=attribute-error
         if credentials.get("pw") is not None or credentials.get("user") is not None:
-            raise_warning(
+            common_utils.raise_deprecation_warning(
                 "Mattermost recently switched to bot accounts. 'user' and 'pw' "
                 "should not be used anymore, you should rather convert your "
                 "account to a bot account and use a token. Password based "
                 "authentication will be removed in a future Rasa Open Source version.",
-                FutureWarning,
+                warn_until_version="3.0.0",
```

Comment:
Should we extract that to a constant? Something like `NEXT_MAJOR_DEPRECATIONS` (assuming that `2.0` is out).
Then it will be super easy to find this things.

Reply:
I can see the pure engineering argument to define the constant, but I don't think it would help finding these things (simply looking for `warn_until_version` would yield the same results).

That being said, we could even go further and:
- define the constant
- make it a default argument to the function: I do not see a use case where we would need to raise the warning until the next next major (e.g. 4.0). WDYT?

Reply:
a546cd93508 WDYT?

Reply:
Sounds good!